.github/workflows/deploy.yml

```
@@ -0,0 +1,322 @@
+jobs:
+  # Pre-deployment quality checks (can be skipped with force_deploy)
+  quality-gate:
+    runs-on: ubuntu-latest
+    if: ${{ !inputs.force_deploy }}
+    
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
```

actionlint (1.7.7)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)